packages/codegen-ui-react/lib/imports/import-collection.ts

```diff
@@ -35,11 +36,51 @@ export class ImportCollection {
 
   importAlias: Map<string, Map<string, string>> = new Map();
 
+  ingestComponentMetadata(componentMetadata: ComponentMetadata) {
+    Object.values(componentMetadata?.componentNameToTypeMap || {}).forEach((value) => this.importedNames.add(value));
+
+    // Add form fields so we dont reuse the identifier
+    if (componentMetadata?.formMetadata) {
+      Object.keys(componentMetadata.formMetadata.fieldConfigs).forEach((value) => this.importedNames.add(value));
+    }
+  }
+
   addMappedImport(importValue: ImportValue) {
     const importPackage = ImportMapping[importValue];
     this.addImport(importPackage, importValue);
   }
 
+  addGraphqlMutationImport(importName: string) {
+    return this.addImport(
+      `${this.importConfig?.rendererConfig?.graphqlConfiguration?.docsFilePath}/mutations`,
+      importName,
+    );
+  }
+
+  addGraphqlQueryImport(importName: string) {
+    return this.addImport(
+      `${this.importConfig?.rendererConfig?.graphqlConfiguration?.docsFilePath}/queries`,
+      importName,
+    );
+  }
+
+  addGraphqlSubscriptionImport(importName: string) {
+    return this.addImport(
+      `${this.importConfig?.rendererConfig?.graphqlConfiguration?.docsFilePath}/subscriptions`,
```

Comment:
Lot of duplication of the majority of this path. Can we clean it up?

Reply:
this has been refactored to reduce duplication